From 8ea7fb2d833e673a13488a9a5301c906b781fc04 Mon Sep 17 00:00:00 2001
Message-Id: <8ea7fb2d833e673a13488a9a5301c906b781fc04.1523725229.git.jan.steffens@gmail.com>
In-Reply-To: <ac8faf2b14ed1c9ce518eddfc9c5098b6f9c74dd.1523725229.git.jan.steffens@gmail.com>
References: <ac8faf2b14ed1c9ce518eddfc9c5098b6f9c74dd.1523725229.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sat, 14 Apr 2018 18:32:44 +0200
Subject: [PATCH 3/5] meson: Define NDEBUG for non-debug-builds

This is for parity with autotools. We were suddenly getting assertion
failures after switching to meson, see [1].

[1]: https://bugs.archlinux.org/task/58218

Signed-off-by: Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
---
 meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index bc1905178d..41a5a0c2ba 100644
--- a/meson.build
+++ b/meson.build
@@ -646,6 +646,8 @@ endif
 # Define DEBUG for debug builds only (debugoptimized is not included on this one)
 if get_option('buildtype') == 'debug'
   pre_args += '-DDEBUG'
+else
+  pre_args += '-DNDEBUG'
 endif
 
 if get_option('shader-cache')
-- 
2.16.2

